Index: audio_encode/tests/tiomxrecorder.h
===================================================================
--- audio_encode.orig/tests/tiomxrecorder.h	2009-09-02 12:29:11.000000000 -0500
+++ audio_encode/tests/tiomxrecorder.h	2009-09-02 12:29:16.000000000 -0500
@@ -12,6 +12,7 @@
 
 #include <error.h>
 #include <OMX_Core.h>
+#include <OMX_Component.h>
 #include <unistd.h>
 #include <getopt.h>
 #include <alsa/asoundlib.h>
