Index: meta/conf/distro/include/poky-fixed-revisions.inc
===================================================================
--- meta/conf/distro/include/poky-fixed-revisions.inc
+++ meta/conf/distro/include/poky-fixed-revisions.inc (working copy)
@@ -177,6 +177,7 @@ SRCREV_pn-libxext ?= "49563f5d76637e2ca28fe0b91ce3114271c0955d"
 SRCREV_pn-libxext ?= "49563f5d76637e2ca28fe0b91ce3114271c0955d"
 SRCREV_pn-xf86-video-intel ?= "87ea531c5dc5b39809395b277c330854aaaaf019"
 SRCREV_pn-xf86-video-intel-dri2 = "3dd7f0f9423bb891bc99cd3b77dc3d57e057a7ef"
+SRCREV_pn-xf86-video-omapfb = "5317aae587a2bf48f07a8c06bfaf7bcbfd23bafc"
 SRCREV_pn-xf86-input-keyboard ?= "3e28d68b50d291938734e9684b8296ca864f3892"
 SRCREV_pn-xf86-input-mouse ?= "ea5cfe804e112f320f14ad896c7802d53551d3e6"
 SRCREV_pn-xf86-input-synaptics ?= "934bc0012f948c52aadc8eda912f7728fb7394a2"
