Index: meta-texasinstruments/packages/extra/xorg-driver-input.inc
===================================================================
--- meta-texasinstruments/packages/extra/xorg-driver-input.inc
+++ meta-texasinstruments/packages/extra/xorg-driver-input.inc (working copy)
@@ -1,2 +1,2 @@
 include xorg-driver-common.inc
-DEPENDS = "randrproto inputproto xserver-xorg xproto"
+DEPENDS = "randrproto inputproto virtual/xserver xproto"
