Index: meta/packages/xorg-xserver/xserver-xf86-lite.inc
===================================================================
--- meta/packages/xorg-xserver/xserver-xf86-lite.inc (revision 5728)
+++ meta/packages/xorg-xserver/xserver-xf86-lite.inc (working copy)
@@ -5,2 +5,2 @@
 PROTO_DEPS = "randrproto renderproto fixesproto damageproto xextproto xproto xf86dgaproto xf86miscproto xf86rushproto xf86vidmodeproto xf86bigfontproto compositeproto recordproto resourceproto videoproto scrnsaverproto evieext trapproto xineramaproto fontsproto kbproto inputproto bigreqsproto xcmiscproto glproto"
-LIB_DEPS = "pixman libxfont xtrans libxau libxext libxdmcp libdrm libxkbfile libpciaccess openssl"
+LIB_DEPS = "pixman libxcalibrate libxfont xtrans libxau libxext libxdmcp libdrm libxkbfile libpciaccess openssl tslib"
@@ -9,2 +9,2 @@
 DEPENDS = "${PROTO_DEPS} ${LIB_DEPS}"
-RDEPENDS = "hal xorg-minimal-fonts"
+RDEPENDS_${PN} += "hal xorg-minimal-fonts"
@@ -16,3 +16,5 @@
                 --disable-afb \
+                 --enable-tslib \
+                 --enable-xcalibrate \
                 --disable-mfb \
                 --disable-dga \
